felix/bpf/polprog/pol_prog_builder_test.go

```diff
@@ -88,15 +90,150 @@ func TestLogActionIgnored(t *testing.T) {
 					Action: "Log",
 				}}},
 			}},
-		}}}, false)
+		}}})
 	Expect(err).NotTo(HaveOccurred())
 
-	pg = NewBuilder(alloc, 1, 2, 3)
+	pg = NewBuilder(alloc, 1, 2, 3, false)
 	noOpInsns, err := pg.Instructions(Rules{
 		Tiers: []Tier{{
 			Name:     "default",
 			Policies: []Policy{},
-		}}}, false)
+		}}})
 	Expect(err).NotTo(HaveOccurred())
 	Expect(noOpInsns).To(Equal(insns))
 }
+
+func TestPolicyDump(t *testing.T) {
+	RegisterTestingT(t)
+	checkLabelsAndComments := func(insns asm.Insns, idx int, expectedString string, matchLabelOrComment string, offset int) {
+		if idx == -1 {
+			idx = getStartofTierInsnIndex(insns)
+			Expect(idx).NotTo(Equal(-1))
+			idx = idx + offset
+		}
+		if matchLabelOrComment == "label" {
+			Expect(checkIfStringExists(insns[idx].Labels, expectedString)).To(BeTrue())
```

Comment:
Think there's a built-in matcher: `Expect(insns[idx].Labels).To(Contain(expectedString))`  You'll get much cleaner errors if that fails.